folium/plugins/timeline.py

```diff
@@ -0,0 +1,313 @@
+from typing import List, Optional, TextIO, Union
+
+from branca.element import MacroElement
+from jinja2 import Template
+
+from folium.elements import JSCSSMixin
+from folium.features import GeoJson
+from folium.folium import Map
+from folium.utilities import JsCode, camelize, get_bounds, parse_options
+
+
+class Timeline(GeoJson):
+    """
+      Creates a layer from GeoJSON with time data to append
+      into a map with Map.add_child.
+
+      To add time data, you need to do one of the following:
+
+      * Add a 'start' and 'end' property to each feature. The start and end
+        can be any comparable item.
+
+
+      Alternatively, you can provide
+      a `get_interval` function.
+
+      * This function should be a JsCode object and take as parameter
+        a GeoJson feature and return a dict containing values for
+        'start', 'end', 'startExclusive' and 'endExcusive' (or false if no
+        data could be extracted from the feature).
+      * 'start' and 'end' can be any comparable items
+      * 'startExclusive' and 'endExclusive' should be boolean values.
+
+      Parameters
+      ----------
+      data: file, dict or str.
+          The geojson data you want to plot.
+
+      auto_play: bool, default True
+          Whether the animation shall start automatically at startup.
+
+      get_interval: JsCode
+          Called for each feature, and should return either a time range for the
+          feature or `false`, indicating that it should not be included in the
+          timeline. The time range object should have 'start' and 'end' properties.
+          Optionally, the boolean keys 'startExclusive' and 'endExclusive' allow the
+          interval to be considered exclusive.
+
+          If `get_interval` is not provided, 'start' and 'end' properties are
+          assumed to be present on each feature.
+
+    Examples
+    --------
+
+    >>> data = requests.get(
+    ...     "https://raw.githubusercontent.com/python-visualization/folium-example-data/main/borders.json"
+    ... ).json()
+    >>> timeline = Timeline(
+    ...     data,
+    ...     style=JsCode(
+    ...         '''
+    ...        function (data) {
+    ...            function getColorFor(str) {
+    ...                // java String#hashCode
+    ...                var hash = 0;
+    ...                for (var i = 0; i < str.length; i++) {
+    ...                    hash = str.charCodeAt(i) + ((hash << 5) - hash);
+    ...                }
+    ...                var red = (hash >> 24) & 0xff;
+    ...                var grn = (hash >> 16) & 0xff;
+    ...                var blu = (hash >> 8) & 0xff;
+    ...                return "rgb(" + red + "," + grn + "," + blu + ")";
+    ...            }
+    ...            return {
+    ...                stroke: false,
+    ...                color: getColorFor(data.properties.name),
+    ...                fillOpacity: 0.5,
+    ...            };
```

Comment:
is this necessary for the example? The example should be the minimum necessary to show how the class works. Or show case some common features. This elaborate custom styling doesn't seem that relevant for learning how the arguments work.

Reply:
It was in the original example for the timeline plugin. I tried the example without colors, but the demo looked horrible.

Reply:
Changed for a simpler example.